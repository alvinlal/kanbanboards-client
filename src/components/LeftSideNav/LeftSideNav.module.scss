.left__side__nav {
  width: 240px;
  height: calc(100vh - 70px);
  padding-top: 40px;
  background-color: $grey-50;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .heading {
    margin-bottom: 16px;

    h6 {
      color: $text-300;
      font-weight: $bold;
    }

    svg {
      color: $text-300;
      transform: rotate(45deg);
      display: none;
      cursor: pointer;
    }
    padding: 0px 24px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .error {
    display: flex;
    align-items: center;
    justify-content: center;
    color: $error-300;
    width: 100%;

    p {
      margin-left: 5px;
    }
  }

  .no__boards {
    margin-left: 25px;
    margin-top: 40px;
  }

  .logout__btn {
    display: none;
    margin-bottom: 10px;
  }

  @media screen and (max-width: $sm) {
    height: 100vh;
    max-height: 100vh;
    position: fixed;
    top: 0px;
    position: absolute;
    width: 100%;
    left: -100%;
    transition: left 0.2s linear;

    .heading {
      margin-top: 40px;
      svg {
        display: block;
      }
    }

    .logout__btn {
      display: flex;
    }
  }
}
