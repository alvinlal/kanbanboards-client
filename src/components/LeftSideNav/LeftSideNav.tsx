/* eslint-disable no-nested-ternary */
import {
  ArrowRightOnRectangleIcon,
  ExclamationTriangleIcon,
  PlusIcon,
} from '@heroicons/react/24/outline';
import { useMemo } from 'react';
import Button from '../Button/Button';
import SkeletonBlock from '../SkeletonBlock/SkeletonBlock';
import BoardList from './components/BoardList/BoardList';
import { useAllBoards } from './hooks/useAllBoards';
import { useLeftSideNav } from './hooks/useLeftSideNav';
import styles from './LeftSideNav.module.scss';

const LeftSideNav: React.FC = () => {
  const { isLeftSideNavOpen, toggleLeftSideNav, logOut } = useLeftSideNav();
  const { allBoards, isLoading, isError } = useAllBoards();

  const LoadingBlocks = useMemo(() => {
    const blocks = [];
    for (let i = 0; i < 6; i += 1) {
      blocks.push(<SkeletonBlock width="200px" height="30px" key={i} />);
    }
    return blocks;
  }, []);

  return (
    <div className={styles.left__side__nav} style={{ left: isLeftSideNavOpen ? 0 : '-100%' }}>
      <div>
        <div className={styles.heading}>
          <h6>All Boards</h6>
          <PlusIcon width={32} height={32} strokeWidth={2} onClick={toggleLeftSideNav} />
        </div>
        {isLoading && <div data-testid="skeleton-blocks"> {LoadingBlocks}</div>}
        {isError && (
          <div className={styles.error}>
            <ExclamationTriangleIcon width={28} height={28} />
            <p>Something went wrong !</p>
          </div>
        )}
        {allBoards ? (
          allBoards.length ? (
            <BoardList allBoards={allBoards} />
          ) : (
            <p className={styles.no__boards}>No boards found</p>
          )
        ) : null}
      </div>
      <div>
        <div className={styles.logout__btn}>
          <Button
            style={{ height: '64px', width: '100%', justifyContent: 'space-between' }}
            onClick={logOut}
          >
            <h6>Logout</h6>
            <ArrowRightOnRectangleIcon width={32} height={32} strokeWidth={2} />
          </Button>
        </div>
        <Button
          style={{ height: '64px', width: '100%', justifyContent: 'space-between', zIndex: 10 }}
        >
          <h6>New Board</h6>
          <PlusIcon width={32} height={32} strokeWidth={2} />
        </Button>
      </div>
    </div>
  );
};

export default LeftSideNav;
