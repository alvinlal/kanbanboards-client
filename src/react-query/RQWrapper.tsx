/* eslint-disable no-console */
// this wrapper is for testing only

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import React from 'react';

const queryClient = new QueryClient({
  logger: {
    log: console.log,
    warn: console.warn,
    error: () => {},
  },
});
export const RQWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);
