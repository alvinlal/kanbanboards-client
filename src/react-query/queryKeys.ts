const queryKeys = {
  user: 'USER',
};

export default queryKeys;
