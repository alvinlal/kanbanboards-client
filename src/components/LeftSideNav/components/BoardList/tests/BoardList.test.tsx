/* eslint-disable no-console */
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import LeftSideNav from '../../../LeftSideNav';

describe('BoardList.tsx', () => {
  const queryClient = new QueryClient({
    logger: {
      log: console.log,
      warn: console.warn,
      error: () => {},
    },
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  beforeEach(() => {
    queryClient.clear();
  });

  it('should render board list correctly', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <MemoryRouter>
          <LeftSideNav />
        </MemoryRouter>
      </QueryClientProvider>
    );

    expect(await screen.findByTestId('board-list-wrapper')).toBeInTheDocument();
  });

  it('should match snapshot', async () => {
    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <MemoryRouter>
          <LeftSideNav />
        </MemoryRouter>
      </QueryClientProvider>
    );

    await waitFor(() => {
      expect(screen.queryByTestId('skeleton-blocks')).not.toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
