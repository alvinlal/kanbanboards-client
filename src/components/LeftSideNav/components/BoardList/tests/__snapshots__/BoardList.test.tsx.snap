// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardList.tsx should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="left__side__nav"
    style="left: -100%;"
  >
    <div>
      <div
        class="heading"
      >
        <h6>
          All Boards
        </h6>
        <svg
          aria-hidden="true"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.5v15m7.5-7.5h-15"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="board__list__wrapper"
        data-testid="board-list-wrapper"
      >
        <div
          class="board__list__container"
          id="#board__list__container"
        >
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               my board 1
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad223casdfdasdfedas392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               test board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad223cgdsdasdged3f392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               this board has a long title and long tilte
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad223cdedaggaerukisd3f392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Lorem ipsum board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2ewrw23cded3giastetf392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               my pro
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad22d3artygsdcded3f392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Another board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2agghreqkur23cded3f392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               my project 7
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad22kljdrg3yeucded3f392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               my project with a very long title
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad22kljdrg3cded3f392afnneye32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Lorem ipsum board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2ewrw23cded3gitf392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Lorem ipsum board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2ewrw23cded3gitf392afasfg"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Lorem ipsum board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2ewrw23cded3gitf3asdg92af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Lorem ipsum board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2ewrw23cded3giwertf392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Lorem ipsum board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2ewrw23cded3giwqrtf392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Lorem ipsum board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2ewrwbqdfg23cded3gitf392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
          <div
            class="board__list__link"
            role="link"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="arrow__right__icon"
              fill="none"
              stroke="currentColor"
              stroke-width="3"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.25 4.5l7.5 7.5-7.5 7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <p>
               Lorem ipsum board
            </p>
            <svg
              aria-hidden="true"
              class="board__options__ellipses"
              data-testid="toggle-dropdown"
              fill="none"
              id="toggle__dropdown__ad2ewrw2hqre3cded3gitf392af32223"
              stroke="currentColor"
              stroke-width="1.5"
              tabindex="0"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="board__list__wrapper"
              data-testid="board-dropdown-wrapper"
              id="board__list__wrapper"
              style="display: none;"
              tabindex="0"
            >
              <div
                class="dropdown"
              >
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Delete
                </div>
                <div
                  class="dropdown__option"
                  role="option"
                  tabindex="0"
                >
                  Duplicate
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="logout__btn"
      >
        <button
          class="button primary "
          style="height: 64px; width: 100%; justify-content: space-between;"
        >
          <h6>
            Logout
          </h6>
          <svg
            aria-hidden="true"
            fill="none"
            height="32"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 9V5.25A2.25 2.25 0 0013.5 3h-6a2.25 2.25 0 00-2.25 2.25v13.5A2.25 2.25 0 007.5 21h6a2.25 2.25 0 002.25-2.25V15m3 0l3-3m0 0l-3-3m3 3H9"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
      <button
        class="button primary "
        style="height: 64px; width: 100%; justify-content: space-between; z-index: 10;"
      >
        <h6>
          New Board
        </h6>
        <svg
          aria-hidden="true"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.5v15m7.5-7.5h-15"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
